src/App.tsx

```diff
@@ -2,15 +2,23 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { useState } from 'react';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
+
+  const addMovie = (movie: Movie) => {
+    setMovies(movies.concat(movie));
```

Comment:
1. use `prev` values
2. you can use spread instead of  `concat`
```suggestion
    setMovies(prevMovies => ({ ...prevMovies, movie }));
```